### PR TITLE
Fix G++ warnings

### PR DESCRIPTION
**Pull request by adrianbroher**
This PR fixes some problematic code, which causes G++ to emit warnings:

* Remove unused and deprecated boost header for endian detection, which emits a warning since boost 1.69.0.
* Remove unreferenced functions.
* Remove unused local variables.
* Catch exceptions by reference in GG Font implementation.
* Add pragma stament block to std_image.h to let GCC ignore multiple "misleading indentations" (talking about pendantic).

The remaining warning about the std::move call in MovableEnvelope was kept as I couldn't yet read into possible consequences of that change.

The warning about violated strict aliasing rules in Effect.cpp require some bigger redesign of the EffectBase interface to be fixed.

### CODE REVIEW

File: GG/src/nanovg/stb_image.h [hunk 1]
```diff
@@ -1375,6 +1378,8 @@ static unsigned char *stbi__convert_format(unsigned char *data, int img_n, int r
          default: STBI_ASSERT(0);
       }
       #undef CASE
+
+      #pragma GCC diagnostic push
```
Comment (o01eg) [inline GG/src/nanovg/stb_image.h]: Shouldn't be there `#pragma GCC diagnostic pop` to pair up with `push` above?

**Review by o01eg — CHANGES_REQUESTED** [review]

File: GG/src/nanovg/stb_image.h (hunk 1 above)
Comment (adrianbroher) [reply GG/src/nanovg/stb_image.h]: Yes it should

Comment (adrianbroher) [reply GG/src/nanovg/stb_image.h]: Fixed via 4dd097c66614fe09845c4b4f4d535a62376e11c0

File: GG/src/Font.cpp [hunk 2]
```diff
@@ -2242,7 +2242,7 @@ void Font::HandleTag(const std::shared_ptr<FormattingTag>& tag, double* orig_col
                     } else {
                         well_formed_tag = false;
                     }
-                } catch (boost::bad_lexical_cast) {
+                } catch (const boost::bad_lexical_cast&) {
```
Comment (geoffthemedio) [inline GG/src/Font.cpp:2245]: could make these just `catch (...)`

Comment (adrianbroher) [reply GG/src/Font.cpp:2245]: It should only catch the `boost::bad_lexical_cast` emitted by the `boost::lexical_cast`, not every exception.

**Pull request merged**